### PR TITLE
IO-1214: Add support to parse field values, make IFieldType.normalize required

### PR DESCRIPTION
**Pull request by jmadson**
https://jira.rediker.com/browse/IO-1214

### CODE REVIEW

File: src/Model/FieldType/BooleanField.test.ts [hunk 1]
```diff
@@ -43,4 +45,9 @@ describe("FieldType: BooleanField", () => {
   it("should be serialized", () => {
     expect(BooleanField.serialize).to.be.true;
   });
+
+  it("normalizes any value into a boolean", () => {
+    const value = "true";
```
Comment (jonsca) [inline src/Model/FieldType/BooleanField.test.ts]: Do we want a couple more test cases to show that the `Boolean` function is not parsing the string for the value "true" but is using the standard rules for 0, "", NaN, etc.?

Comment (jmadson) [reply src/Model/FieldType/BooleanField.test.ts]: Good idea!

File: src/Model/FieldType/NumberField.test.ts [hunk 2]
```diff
@@ -42,4 +42,9 @@ describe("FieldType: NumberField", () => {
   it("should be serialized", () => {
     expect(NumberField.serialize).to.be.true;
   });
+
+  it("normalizes any value into a string", () => {
```
Comment (jonsca) [inline src/Model/FieldType/NumberField.test.ts]: This test is misnamed

File: src/Model/FieldType/EmailField.ts [hunk 3]
```diff
@@ -1,9 +1,7 @@
 import { IFieldType } from "./IFieldType";
-import { isString } from "validate.js";
+import { StringField } from "./StringField";
 
 export const EmailField: IFieldType<string> = {
-  serialize: true,
+  ...StringField,
   defaultValidationRules: { email: true },
-  defaultValue: "",
-  isValidType: (value) => value == null || isString(value),
```
Comment (jonsca) [inline src/Model/FieldType/EmailField.ts]: Where is this getting its `normalize` function from?

Comment (jmadson) [reply src/Model/FieldType/EmailField.ts]: It was updated to use the ES6 `spread` operator, so the properties from the `StringField` are being copied onto the object, as if `EmailField` were extending `StringField`.

Comment (jonsca) [reply src/Model/FieldType/EmailField.ts]: Ah, makes sense.  No change needed, then.

**Comment by jonsca** [comment]
385 tests pass for me locally.  Just those few small changes and it should be good to go.

**Comment by jmadson** [comment]
Thanks @jonsca ! I just pushed the requested changes.

**Comment by jonsca** [comment]
392 tests pass locally and all changes were made.

**Pull request merged**